### PR TITLE
Exporting of user data CSV trips over UnicodeDecodeError

### PR DESCRIPTION
**Pull request by smn**
```

Traceback (most recent call last):

  File "/var/praekelt/vumi-go/ve/lib/python2.6/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = callback(request, *callback_args, **callback_kwargs)

  File "/var/praekelt/vumi-go/ve/src/vxpolls/vxpolls/djdashboard/views.py", line 99, in export_users
    results = poll.results_manager.get_users_as_csv(poll_id)

  File "/var/praekelt/vumi-go/ve/src/vumi/vumi/persist/redis_base.py", line 97, in wrapper
    return manager.call_decorator(func)(self, *args, **kw)

  File "/var/praekelt/vumi-go/ve/src/vumi/vumi/utils.py", line 241, in wrapped
    result = gen.send(result)

  File "/var/praekelt/vumi-go/ve/src/vxpolls/vxpolls/results.py", line 205, in get_users_as_csv
    writer.writerow(self.encode_as_utf8(data))

  File "/var/praekelt/vumi-go/ve/src/vxpolls/vxpolls/results.py", line 187, in encode_as_utf8
    for k, v in dictionary.items())

  File "/var/praekelt/vumi-go/ve/src/vxpolls/vxpolls/results.py", line 187, in <genexpr>
    for k, v in dictionary.items())

UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 8: ordinal not in range(128)
```

### CODE REVIEW

**Comment by smn** [comment]
This is waiting on praekelt/vumi#357 to land.

File: vxpolls/results.py [hunk 1]
```diff
@@ -152,8 +156,11 @@ def get_results_for_question(self, collection_id, question):
         if answers:
             results = []
             for answer in answers:
-                result = yield self.r_server.hget(results_key, answer)
-                results.append((answer, int(result or 0)))
+                if (yield self.r_server.hexists(results_key, answer)):
+                    result = yield self.r_server.hget(results_key, answer)
```
Comment (jerith) [inline vxpolls/results.py:160]: Why the change to two calls here?

**Comment by jerith** [comment]
:+1:, although I'm interested in the reason for the change I commented on in the diff.

File: vxpolls/results.py (hunk 1 above)
Comment (smn) [inline vxpolls/results.py:160]: Before the change in FakeRedis's char handling `hset()` left `None` as `None`, now it'll turn it to `"None"` which the `int(result or 0)` is not happy with.

Comment (jerith) [inline vxpolls/results.py:160]: Can we fix this properly in FakeRedis? I assume real redis doesn't do that...

Comment (smn) [inline vxpolls/results.py:160]: No, because we didn't do any type conversion in `FakeRedis`, `None` remained `None` when we stored it in `FakeRedis` -- real Redis stores it as `"None"` when we try and store `None` -- which is what it does now.
